### PR TITLE
Ability to use local SQLite database, automatic database structure setup, Discord bot disabling

### PR DESCRIPTION
**Pull request by Eric5766**
Added:

- BotSettings 'DiscordBotEnabled' & 'DatabaseType' (sqlite or mssql)
- DatabaseWrapper class that handles all requests to save/retrieve data depending on DatabaseType, with flexibility to add new storage types.
- Sql Schema files for both mssql and sqlite which automatically set up necessary tables in database.

General flow:
- User sets up database type in settings
- On initialisation, depending on database type, new SQLite database is created and/or local .SQL script run to create all necessary tables
- On data request, repository passes arguments into corresponding database wrapper method.
- Database wrapper methods contain logic that retrieves data based on database type and passes it back through in expected format.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -279,4 +279,5 @@ main.css
 main.min.css
 *.min.js
 
-Bot
\ No newline at end of file
+Bot
+/BloogBot/UI/App.xaml.cs
```
Comment (DrewKestell) [inline .gitignore:283]: Why are we excluding this from the git repo? I think this needs to be included.

File: BloogBot/AI/SharedStates/EquipBagsState.cs [hunk 2]
```diff
@@ -23,7 +23,7 @@ public void Update()
         {
             if (ObjectManager.Player.IsInCombat)
             {
-                botStates.Pop();
+                //botStates.Pop();
```
Comment (DrewKestell) [inline BloogBot/AI/SharedStates/EquipBagsState.cs:26]: I don't think this change is safe. If the player is in combat, but has an empty bag slot, and a bag in their inventory that needs to be equipped, you'd get stuck in this state forever returning.

File: BloogBot/AI/SharedStates/GatherObjectState.cs [hunk 3]
```diff
@@ -8,13 +8,13 @@ namespace BloogBot.AI.SharedStates
...
         readonly Stack<IBotState> botStates;
-        readonly WoWGameObject target;
+        readonly dynamic target;
         readonly LocalPlayer player;
         readonly int initialCount = 0;
 
         readonly int startTime = Environment.TickCount;
 
-        internal GatherObjectState(Stack<IBotState> botStates, IDependencyContainer container, WoWGameObject target)
+        internal GatherObjectState(Stack<IBotState> botStates, IDependencyContainer container, dynamic target)
```
Comment (DrewKestell) [inline BloogBot/AI/SharedStates/GatherObjectState.cs:17]: Why make this `dynamic`? I don't see any changes that would require this.

**Comment by Eric5766** [comment]
Ahh, sorry I'm new to using these tools.  I was only meant to include the
database stuff in the request, I didn't realise it'd include all of my
recent messing about as well.

I'll go back and clean it up, ignore for now I guess

**Pull request closed without merging**